drivers/clock_control/clock_control_esp32.c

```diff
@@ -32,6 +32,17 @@
 #define DT_CPU_COMPAT espressif_riscv
 #include <zephyr/dt-bindings/clock/esp32c3_clock.h>
 #include <esp32c3/rom/rtc.h>
+#elif CONFIG_SOC_SERIES_ESP32C6
+#define DT_CPU_COMPAT espressif_riscv
+#include <zephyr/dt-bindings/clock/esp32c6_clock.h>
+#include <soc/lp_clkrst_reg.h>
+#include <soc/regi2c_dig_reg.h>
+#include <regi2c_ctrl.h>
+#include <esp32c6/rom/rtc.h>
+#include <soc/dport_access.h>
+#include <hal/clk_tree_ll.h>
+#undef MHZ
```

Comment:
This sounds wrong. Can this be removed and reworked?

Reply:
This was discussed with Lucas. Not a nice workaround but otherwise would require changing many HAL files, thus affecting other SoC's.

Reply:
Please, fix HAL, this is not allowed per guidelines.